### ISSUE TITLE
Use TASKs to speed up CI

### ISSUE THREAD

**Issue by jameslamb** [jameslamb/doppel-cli#37]
Would be cool to try implementing this trick that LightGBM uses:

https://github.com/Microsoft/LightGBM/blob/master/.travis.yml#L15

Basically, you define an environment variable called `TASK`, run a matrix of builds (one per task), then all the builds share a common `do_stuff.sh` that has code in it like "if TASK = 'whatever' do something".

In this project, I could see value in parallel tasks for:

* running coverage / unit tests
* running integration tests
* testing for documentation issues
* checking for code style issues with `pycodestyle`
* running integration tests (literally using `uds-cli`, not the underlying python code)

**Comment by jameslamb** [comment]
Environment setup is by far the largest cost for this project, so a strategy that involves setting up _multiple_ VMs on Travis doesn't seem like a good one.

Builds for `doppel-cli` are currently taking around 5 minutes and 4.5 minutes of that is just installing stuff with `conda`.

Going to remove this from 0.3.0 milestone.

It _could_ still speed up builds if the CI got more complicated (like if a bunch of R and Python smoke tests are added), but this is low priority for now.

**Comment by jameslamb** [comment]
Thinking about this some more...this actually COULD speed up CI. Things like code linting don't need to be repeated on every combination of operating system and R version and Python version. It could be valuable to have one "main" build that does a bunch of stuff and then only test unit tests plus installation on the other OSes.
